packages/twenty-server/src/workspace/workspace-query-runner/workspace-query-runner.service.ts

```diff
@@ -63,14 +65,22 @@ export class WorkspaceQueryRunnerService {
     args: FindManyResolverArgs<Filter, OrderBy>,
     options: WorkspaceQueryRunnerOptions,
   ): Promise<IConnection<Record> | undefined> {
-    const { workspaceId, objectMetadataItem } = options;
+    const { workspaceId, userId, objectMetadataItem } = options;
     const start = performance.now();
 
     const query = await this.workspaceQueryBuilderFactory.findMany(
       args,
       options,
     );
 
+    await this.workspacePreQueryHookService.executePreHooks(
```

Comment:
@Weiko Maybe we can attach the hook service with a decorator too ?

Reply:
Yes!